### PR TITLE
Implemented /consensus endpoint

### PR DESCRIPTION
**Pull request by mxmauro**
## Summary

Implemented the `/consensus` endpoint. It has an optional `version` query parameter. If not provided, all protocol consensus details are given, else the one that matches the specified one.

## Test Plan

Run:

`curl -H "X-Algo-API-Token: {token}" http://{host}:{port}/consensus`

`curl -H "X-Algo-API-Token: {token}" http://{host}:{port}/consensus?version=v9`

### CODE REVIEW

**Comment by mxmauro** [comment]
This modification aims to implement the feature requested in https://github.com/algorand/go-algorand/issues/2148

**Comment by mxmauro** [comment]
Hi @winder , an assistance on this will be appreciated. I had problems with the dependencies related to the makefile script that processes swagger json files and generated ones and the build is failing in something I don't completely understand. If you can give me some guidance I can try to fix myself.

Thanks!

**Comment by winder** [comment]
> Hi @winder , an assistance on this will be appreciated. I had problems with the dependencies related to the makefile script that processes swagger json files and generated ones and the build is failing in something I don't completely understand. If you can give me some guidance I can try to fix myself.
> 
> Thanks!

What error are you seeing?

**Comment by mxmauro** [comment]
First when I run the `Makefile` at `daemon/algod/api` the command below fails to complete.

`GO111MODULE=on go get -u "github.com/algorand/oapi-codegen/...@v1.3.5-algorand5"`

But, for some reason, the `oapi-codegen` binary was built (I think some of the dependencies are not build).

I did the modifications to `algod.oas2.json` but they are not propagated to other files. Also tried running `make rebuild_swagger` but still nothing changed (swagger files were untouched).

So I decided to modify files manually and regenerate files including the swagger data embedded into `bundledSpecInject.go`. All compiles and seems fine but the coverage test ran here fails and I'm unable to fully understand the reason.

**Comment by winder** [comment]
@mxmauro try running `scripts/buildtools/install_buildtools.sh` first to install the dependencies needed to rebuild swagger.

**Comment by mxmauro** [comment]
@winder I remember I did it but tried again and after re-ran `make rebuild_swagger`. Because some files were modified, commited and hoping it works.

**Comment by mxmauro** [comment]
@winder Finally! Thanks for the assistance.

BTW: Unrelated but this is what I get when retrieving oapi-codegen:

```# github.com/algorand/oapi-codegen/pkg/util
../../../../../pkg/mod/github.com/algorand/oapi-codegen@v1.3.5-algorand5/pkg/util/loader.go:13:37: undefined: openapi3.Swagger
../../../../../pkg/mod/github.com/algorand/oapi-codegen@v1.3.5-algorand5/pkg/util/loader.go:19:15: undefined: openapi3.Swagger
../../../../../pkg/mod/github.com/algorand/oapi-codegen@v1.3.5-algorand5/pkg/util/loader.go:24:18: undefined: openapi3.NewSwaggerLoader
../../../../../pkg/mod/github.com/algorand/oapi-codegen@v1.3.5-algorand5/pkg/util/loader.go:26:14: undefined: openapi3.Swagger
...and more
```

File: daemon/algod/api/server/common/handlers.go [hunk 1]
```diff
@@ -48,6 +50,60 @@ func GenesisJSON(ctx lib.ReqContext, context echo.Context) {
...
+			w.Write([]byte(fmt.Sprintf("Unknown parameter detected: %s", name)))
+			return
+		}
+	}
+
+	version := context.QueryParam("version")
+	if len(version) == 0 {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write(protocol.EncodeJSON(config.Consensus))
```
Comment (algorandskiy) [inline daemon/algod/api/server/common/handlers.go:93]: maybe return not everything but keys (consensus versions) ?

**Review by algorandskiy — COMMENTED** [review]
Thank you for the submission. This needs to be implemented as v2 handler since we consider v1 REST API deprecating one.

File: daemon/algod/api/server/common/handlers.go (hunk 1 above)
Comment (mxmauro) [reply daemon/algod/api/server/common/handlers.go:93]: May be, but also may be it won't be necessary to expose the whole protocol information but only useful stuff.

**Comment by bbroder-algo** [comment]
This PR has kind of slid out of focus so closing it but if still desired we could track it as an issue for reprioritization

**Pull request closed without merging**